packages/pigment-css-react/src/generateAtomics.js

```diff
@@ -11,37 +11,73 @@ export function generateAtomics() {
  * @property {Object.<string, Object.<string, Object.<string, string>>>} styles
  * @property {Object.<string, string[]>} shorthands
  * @property {string[]} conditions
+ * @property {string} defaultCondition
+ * @property {string[]} unitless
+ * @property {string} multiplier
  */
 
 /**
+ * Given a mapping of atomic classes to css properties for each breakpoint, this function
+ * returns all the classnames associated with the runtime values.
+ *
  * @param {RuntimeConfig} runtimeConfig
  */
-export function atomics({ styles, shorthands, conditions }) {
-  function addStyles(cssProperty, values, classes) {
+export function atomics({
+  styles,
+  shorthands,
+  conditions,
+  defaultCondition,
+  unitless,
+  multiplier = 1,
+}) {
+  function addStyles(cssProperty, propertyValue, classes, inlineStyle) {
     const styleClasses = styles[cssProperty];
     if (!styleClasses) {
       return;
     }
-    if (typeof values === 'string') {
-      classes.push(styleClasses[values].$$default);
-    } else if (Array.isArray(values)) {
-      values.forEach((value, index) => {
-        classes.push(styleClasses[value][conditions[index]]);
+
+    function handlePrimitive(value, breakpoint = '$$default') {
+      if (!(value in styleClasses)) {
+        const keys = Object.keys(styleClasses);
+        if (keys.length !== 1) {
+          return;
+        }
+        const key = keys[0];
+        const styleValue = typeof value === 'number' ? value * multiplier : value;
+        classes.push(styleClasses[key][breakpoint]);
+        inlineStyle[`${key}_${breakpoint === '$$default' ? defaultCondition : breakpoint}`] =
+          unitless.includes(cssProperty) ? styleValue : `${styleValue}px`;
+      } else {
+        classes.push(styleClasses[value][breakpoint]);
+      }
+    }
+
+    if (typeof propertyValue === 'string' || typeof propertyValue === 'number') {
+      handlePrimitive(propertyValue);
+    } else if (Array.isArray(propertyValue)) {
+      propertyValue.forEach((value, index) => {
+        const breakpoint = conditions[index];
+        if (!breakpoint) {
+          return;
+        }
+        handlePrimitive(value, conditions[index]);
       });
     } else {
-      Object.keys(values).forEach((condition) => {
-        const propertyClasses = styleClasses[values[condition]];
+      Object.keys(propertyValue).forEach((condition) => {
+        const propertyClasses = styleClasses[propertyValue[condition]];
         if (!propertyClasses) {
+          handlePrimitive(propertyValue[condition], condition);
           return;
         }
-        classes.push(propertyClasses[condition]);
+        handlePrimitive(propertyClasses[condition], condition);
```

Comment:
It's working correct, see the test `should work with shorthands`.

Reply:
The test works correctly because it uses an array for the direction value, which is handled here: https://github.com/mui/pigment-css/blob/e3408002b3c155f915b5b26f76e343202dee254a/packages/pigment-css-react/src/generateAtomics.js#L52

The test fails if you provide `direction: { xs: 'row', sm: 'column }`, which should also be supported, and it's handled in the line we're discussing.

So besides changing this line we should add a test with all supported formats.

Reply:
You are right. I pushed a fix with all the test cases that also cover `null`, `undefined` from the values.